sql/parser/select.go

```diff
@@ -63,7 +64,13 @@ func (p *Parser) parseResultFields() ([]query.ResultField, error) {
 
 	// Parse remaining (optional) result fields.
 	for {
-		if tok, _, _ := p.ScanIgnoreWhitespace(); tok != scanner.COMMA {
+		tok, _, _ := p.ScanIgnoreWhitespace()
+		if tok != scanner.COMMA {
+			p.Unscan()
+			return rfields, nil
+		}
+		tok, _, _ = p.ScanIgnoreWhitespace()
+		if tok != scanner.DOT {
```

Comment:
This would allow users to write `SELECT foo, . foo FROM table`, could you explain your reasoning behind that?